crane/container.go

```diff
@@ -1035,12 +1193,28 @@ func (c *container) Exec(cmds []string) {
 	}
 	for _, name := range runningInstances {
 		args := []string{"exec"}
-		if c.ExecParams().Interactive {
+		if c.ExecParams().Detach || c.RunParams().Detach {
+			args = append(args, "--detach")
+		}
+		if len(c.ExecParams().DetachKeys()) > 0 {
+			args = append(args, "--detach-keys", c.ExecParams().DetachKeys())
+		} else if len(c.RunParams().DetachKeys()) > 0 {
+			args = append(args, "--detach-keys", c.RunParams().DetachKeys())
+		}
+		if c.ExecParams().Privileged || c.RunParams().Privileged {
```

Comment:
Yea ... I think this is such an edge case that it's not worth it. I'm fine with the proposed behaviour.